Code/Core/Env/Assert.cpp

```diff
@@ -85,7 +85,7 @@ bool IsDebuggerAttached()
         const uint32_t BUFFER_SIZE( 4096 );
         char buffer[ BUFFER_SIZE ];
         #if defined( __APPLE__ ) || defined( __LINUX__ )
-            sprintf( buffer,
+            snprintf( buffer, BUFFER_SIZE,
```

Comment:
These snprintf changes seem ok but unrelated to file permissions?

Reply:
Removed. @Leandros needed these changes to build for OSX, but I'll let you guys work that out.